src/datasources/asset/components/AssetQueryEditor.tsx

```diff
@@ -80,42 +106,67 @@ export function AssetQueryEditor({ query, onChange, onRunQuery, datasource }: Pr
 
   return (
     <div style={{ position: 'relative' }}>
-      <InlineField label="Workspace"
-                   tooltip={tooltips.workspace[EntityType.Asset]}
-                   labelWidth={22}>
-        <Select
-          isClearable
-          isLoading={workspaces.loading}
-          onChange={onWorkspaceChange}
-          options={workspaces.value}
-          placeholder="Any workspace"
-          value={query.workspace}
-        />
-      </InlineField>
-      <InlineField label="Systems"
-                   tooltip={tooltips.system[EntityType.Asset]}
-                   labelWidth={22}>
-        <MultiSelect
-          isClearable
-          allowCreateWhileLoading
-          options={loadMinionIdOptions()}
-          isValidNewOption={isValidId}
-          onChange={handleMinionIdChange}
-          placeholder="Select systems"
-          width={85}
-          value={query.minionIds.map(toOption)}
+      <InlineField label="Query type" labelWidth={22}>
+        <RadioButtonGroup
+          options={queryTypeOptions}
+          onChange={handleQueryTypeChange}
+          value={query.queryKind}
         />
       </InlineField>
-      <FloatingError message={errorMsg}/>
+      {query.queryKind === AssetQueryType.CalibrationForecast && (
+        <>
```

Comment:
I'd be great if we could split this into two separate react components to handle the code in a nicer way.

Now we are basically just mixing everything together and it will become a hell to maintain.